doc/SONIC_Test_Ingress_Discard_All_HLD.md

```diff
@@ -0,0 +1,606 @@
+Test objective
+
+Verify packet drops when SMAC and DMAC are equal.
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = 00:00:00:00:01:02
+  src = 00:00:00:00:01:02
+  type = 0x800
+...
+```
+
+Test steps
+- PTF host will send IP packet specifying identical src and dst MAC.
+- When packet reaches SONIC DUT, it should be dropped by "SAI_IN_DROP_REASON_SMAC_EQUALS_DMAC"
+- Verify "ingress_discard_all" counter increment
+
+#### Test case #2
+Test objective
+
+Verify packet drops when packet VLAN ID does not match ingress port VLAN ID
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x8100
+  vid = 2
+###[ IP ]###
+    version = 4  
+    ttl = [auto]
+    proto = tcp  
+    src = 10.0.0.2
+    dst = [get_from_route_info]
+...
+```
+
+Test steps
+- PTF host will send IP packet specifying VID different then port VLAN
+- When packet reaches SONIC DUT, it should be dropped by "SAI_IN_DROP_REASON_VLAN_TAG_NOT_ALLOWED"
+- Verify "ingress_discard_all" counter increment
+
+#### Test case #3
+Test objective
+
+Verify packet with SMAC Multicast drops
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = 01:00:5e:00:01:02
+  type = 0x800
+###[ IP ]###
+    version = 4  
+    ttl = [auto]
+    proto = tcp  
+    src = 10.0.0.2
+    dst = [get_from_route_info]
+...
+```
+
+Test steps
+- PTF host will send IP packet specifying multicast SMAC.
+- When packet reaches SONIC DUT, it should be dropped by "SAI_IN_DROP_REASON_SMAC_MULTICAST"
+- Verify "ingress_discard_all" counter increment
+
+#### Test case #4
+Test objective
+
+Verify packet with reserved DMAC drops
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = 01:80:c2:00:00:09
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = 10.0.0.2
+    dst = [get_from_route_info]
+...
+```
+
+Test steps
+- PTF host will send IP packet specifying reserved DMAC
+- When packet reaches SONIC DUT, it should be dropped by "SAI_IN_DROP_REASON_DMAC_RESERVED"
+- Verify "ingress_discard_all" counter increment
+
+#### Test case #5
+Test objective
+
+Verify packet drops by loop-back filter
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = [auto]
+    dst = [known_bgp_neighboar_ip]
+...
+```
+
+Test steps
+- PTF host will send IP packet specifying DST IP of VM host. Port to send is port which IP interface is in VM subnet.
+- When packet reaches SONIC DUT, it should be dropped by "SAI_IN_DROP_REASON_L2_LOOPBACK_FILTER"
+- Verify "ingress_discard_all" counter increment
+
+#### Test case #6
+Test objective
+
+Verify packet which exceed router interface MTU (for IP and/or MPLS packets) drops
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+...
+###[ TCP ]###  
+    sport = [auto]
+    dport = [auto]
+    data = [max_mtu + 1]
```

Comment:
You need to check if testbed trunk port has mtu greater than 9100 in order to run this test on default configuration

Reply:
Added note about this in test case description.